### ISSUE TITLE
MissingValueException while using pmml from R

### ISSUE THREAD

**Issue by edenwind** [jpmml/jpmml-evaluator#126]
Hi there
I am facing a problem while using a model trained in R. It throws a MissingValueException: The value for field "label" is not defined. After checking every parts in my pmml file which contains "label", i cannot find any abnormal parts. The type of model is xgboost (https://github.com/dmlc/xgboost). By the way, the data which was used to train this model in R has 56 columns and contains a column named "label", the data which is using now for test purpose has all same 55 columns as the training data except for the "label" column, but i don't think it's the problem here.

The scala version is 2.11.8, and the pmml-evaluator version is 1.4.3

Because of the file size limit, i have put pmml file and data file  into the google drive and here is the shared link
pmml file: [https://drive.google.com/open?id=1QOi-G-sfKEh5yx0r21-aavgSM0nkkqUy](url)
data file: [https://drive.google.com/open?id=1B7PExH17ZPK2UnmmQCcLOFC6oefdolT3](url)

**The Exception it throws:**
```
Exception in thread "main" org.jpmml.evaluator.MissingValueException: The value for field "label" is not defined
	at org.jpmml.evaluator.OutputUtil.evaluate(OutputUtil.java:144)
	at org.jpmml.evaluator.mining.MiningModelEvaluator.evaluate(MiningModelEvaluator.java:250)
	at org.jpmml.evaluator.mining.MiningModelEvaluator.evaluate(MiningModelEvaluator.java:209)
	at ModelInvoker.invoke(ModelInvoker.java:44)
	at Simple.main(Simple.java:21)
```
**Here is my test code**

**test class:**

```Java
import org.dmg.pmml.FieldName;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Simple {

    public static void main(String[] args) throws IOException {

        String modelPath = "pmml/80.pmml";
        String dataPath = "file/data.csv";
        try {
            ModelInvoker invoker = new ModelInvoker(modelPath);
            List<Map<FieldName, Object>> data = readData(dataPath);
            for (Map<FieldName, Object> elem : data) {
                Map<FieldName, ?> result = invoker.invoke(elem);
                System.out.println(result);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private static List<Map<FieldName,Object>> readData(String path) throws IOException{

        BufferedReader br = new BufferedReader(new FileReader(path));
        String[] header = br.readLine().split(",");

        List<Map<FieldName,Object>> data = new ArrayList<>();

        String line = null;
        while((line = br.readLine()) != null){

            Map<FieldName,Object> map = new HashMap<FieldName, Object>();

            String[] dataLine = line.split(",");
            assert (dataLine.length == header.length);
            for(int i=0; i<dataLine.length; i++){
                if(dataLine[i].equalsIgnoreCase("null")) map.put(new FieldName(header[i]), null);
                else map.put(new FieldName(header[i]), dataLine[i]);
            }
            data.add(map);
        }
        return data;
    }
}

```

**ModelInvoker class:**

```Java
import org.dmg.pmml.FieldName;
import org.dmg.pmml.PMML;
import org.jpmml.evaluator.ModelEvaluator;
import org.jpmml.evaluator.ModelEvaluatorFactory;
import org.jpmml.model.PMMLUtil;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.Map;

public class ModelInvoker {

    private ModelEvaluator modelEvaluator; 
    
    public ModelInvoker(String path) throws IOException {

        PMML pmml = null;
        InputStream in = null;
        try {
            if (path != null) {
                File file  = new File(path);
                in = new FileInputStream(file);
                pmml = PMMLUtil.unmarshal(in);
            }
            this.modelEvaluator = ModelEvaluatorFactory.newInstance().newModelEvaluator(pmml);
        }
        catch (Exception e) {
            pmml = null;
        }
        finally {
            try {
                in.close();
            }
            catch (IOException ioe) {
                System.out.println("Failed to close file: "+ioe);
            }
        }
        this.modelEvaluator.verify();
    }

    public Map<FieldName, ?> invoke(Map<FieldName, Object> paramsMap) {
        return this.modelEvaluator.evaluate(paramsMap);
    }
}
```
Could you please check why this problem happens and how to fix it?
Thank you very much!

**Comment by vruusmann** [comment]
> Could you please check why this problem happens and how to fix it?

**Diagnosis** - The `pmml` package that you've used to export your XGBoost model has generated invalid/incorrect PMML markup.

**Cure** - Switch from the `pmml` package to the `r2pmml` package:
https://github.com/jpmml/r2pmml

The `r2pmml` package generates correct PMML markup (which is also considerably smaller than the above `pmml` package-generated incorrect PMML markup - go figure!).

**Comment by vruusmann** [comment]
PS: your Java code appears to be correct. Simply switch to a better exporter tool, and everything will work as advertised.
